src/index.ts

```diff
@@ -27,34 +19,32 @@ class Hold<A> extends MulticastSource<A> {
 
     // This also adds the new sink to the internal sinks array.
     // At this point, the sink is in both this.sinks and this.pendingSinks,
-    // and later, _flushPending will remove it from this.pendingSinks.
+    // and later, flushPending will remove it from this.pendingSinks.
     return super.run(sink, scheduler)
   }
 
-  _hasValue (): boolean {
-    return this.held !== undefined
-  }
-
-  _hasSinks (): boolean {
-    return this.sinks.length > 0
-  }
-
-  _shouldScheduleFlush (): boolean {
-    return this._hasValue() && this._hasSinks()
-  }
-
   dispose (): void {
     this._cancelTask()
     return super.dispose()
   }
 
   event (time: Time, value: A): void {
-    this._flushPending(time)
+    this.flushPending(time)
     this.held = { value }
     super.event(time, value)
   }
 
-  _flushPending (time: Time): void {
+  end (time: Time): void {
+    this.flushPending(time)
+    super.end(time)
+  }
+
+  error (time: Time, err: Error): void {
+    this.flushPending(time)
+    super.error(time, err)
+  }
+
+  flushPending (time: Time): void {
```

Comment:
Since this was "private by convention" previously, let's make it `private`.  It's not dangerous per se, but it wasn't intended to be a part of `Hold`'s public API.  If there are concrete use cases for making it public, we could consider that in the future.

Reply:
It is used in `HoldTask#run` so it should be public. Anyway `Hold` interface is not exported in typings so it's pretty internal.

Reply:
Ah, yes, I see.  It's effectively "package private".  Thanks for explaining.